### PR TITLE
Fix issue #8197: Add unit tests for isLikelyDirectory

### LINKED ISSUE #8197: Add unit tests for isLikelyDirectory
**What problem or use case are you trying to solve?**

Find the tests we have for path-component.tsx and familiarize yourself with how we unit test frontend code.

Then add unit tests for isLikelyDirectory().

Please make sure to follow the patterns in the project.

### PR DESCRIPTION
**Pull request by openhands-agent**
This pull request fixes #8197.

The issue has been successfully resolved. The requested task was to add unit tests for the isLikelyDirectory() function following existing project patterns, and this has been fully implemented:

1. A new test file path-component.test.tsx was created in the appropriate test directory
2. Comprehensive test cases were added covering all major scenarios:
   - Empty paths
   - Paths ending with forward slashes
   - Paths ending with backslashes
   - Paths without extensions
   - Paths ending with dots
   - Paths with file extensions
3. The tests follow the project's existing patterns using vitest
4. The isLikelyDirectory function was properly exported from path-component.tsx to make it testable
5. Each test case has clear, descriptive names and tests a specific behavior

The changes provide complete test coverage for the isLikelyDirectory() function's expected behavior, ensuring it correctly identifies directory-like paths across different formats and edge cases. The implementation matches standard unit testing practices and integrates properly with the existing codebase.

Automatic fix generated by [OpenHands](https://github.com/All-Hands-AI/OpenHands/) 🙌

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:5525e87-nikolaik   --name openhands-app-5525e87   docker.all-hands.dev/all-hands-ai/openhands:5525e87
```

### CODE REVIEW

File: frontend/__tests__/components/features/chat/path-component.test.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  it("should return true for paths without extension", () => {
+    expect(isLikelyDirectory("/path/to/dir")).toBe(true);
+    expect(isLikelyDirectory("dir")).toBe(true);
+  });
+
+  it("should return true for paths ending with dot", () => {
+    expect(isLikelyDirectory("/path/to/dir.")).toBe(true);
+    expect(isLikelyDirectory("dir.")).toBe(true);
+  });
```
Comment (enyst) [inline frontend/__tests__/components/features/chat/path-component.test.tsx:27]: I didn't expect this one (nor windows)

Should we return false here? 🤔

Comment (rbren) [reply frontend/__tests__/components/features/chat/path-component.test.tsx:27]: I think the logic here was trying to detect `.` and `..`. So  yeah this one is probably false ideally

**Review by rbren — APPROVED** [review]
Thank you!

File: frontend/__tests__/components/features/chat/path-component.test.tsx (hunk 1 above)
Comment (enyst) [reply frontend/__tests__/components/features/chat/path-component.test.tsx:27]: @openhands-agent-exp Above this line, we have this test:
`it("should return true for paths ending with dot", ()`. Find it and read it.

Fix the actual implementation of isLikelyDirectory() to return false in this case, and fix the test to test for false, instead of true.

**Comment by openhands-agent** [comment]
[OpenHands](https://github.com/All-Hands-AI/OpenHands) started fixing the pr! You can monitor the progress [here](https://github.com/All-Hands-AI/OpenHands/actions/runs/14776200731).

**Comment by openhands-agent** [comment]
Overview:
The changes successfully resolved the issue by:
1. Removing logic that incorrectly treated paths ending in dots as directories
2. Updating tests to verify paths ending in dots now return false

Status: ✅ All issues appear to be fully resolved
- Implementation and tests are now consistent
- Changes align with the intended behavior for handling paths ending in dots
- No remaining issues identified

File: frontend/__tests__/components/features/chat/path-component.test.tsx (hunk 1 above)
Comment (openhands-agent) [reply frontend/__tests__/components/features/chat/path-component.test.tsx:27]: Openhands fix success summary


The feedback has been successfully incorporated through two key changes:

1. The implementation in path-component.tsx was modified to remove the `|| lastPart.endsWith(".")` condition, which means paths ending with a dot are no longer considered directories. This directly addresses the feedback questioning whether paths ending in dots should return false.

2. The test case was appropriately updated to match this new behavior - both the test name and expectations were changed from "should return true" to "should return false" for paths ending with dot, and the assertions now expect false for "/path/to/dir." and "dir."

The changes are logically consistent with each other - the implementation change ensures paths ending in dots return false, and the test now verifies this behavior. This matches the feedback's suggestion that paths ending in dots should return false, particularly since the original logic was only trying to detect "." and ".." special directory cases.

**Review by raymyers — APPROVED** [review]
Thanks for this!

File: frontend/src/components/features/chat/path-component.tsx [hunk 2]
```diff
@@ -23,8 +23,9 @@ const isLikelyDirectory = (path: string): boolean => {
   if (path.endsWith("/") || path.endsWith("\\")) return true;
   // Check if path has no extension (simple heuristic)
   const lastPart = path.split(/[/\\]/).pop() || "";
-  // If the last part has no dots or ends with a dot, it's likely a directory
-  return !lastPart.includes(".") || lastPart.endsWith(".");
+  // If the last part has no dots, it's likely a directory
+  // Note: Paths ending with a dot are not considered directories
```
Comment (enyst) [inline frontend/src/components/features/chat/path-component.tsx:27]:
```suggestion
```
